test/statement.test.ts

```diff
@@ -0,0 +1,532 @@
+    });
+
+    test("when trying to extract a missing column", async function () {
+      // In the following, if we aren't using generics, typescript would catch that missing isn't valid
+      // We use "any" as the type just to test passing invalid colum names
+      await rejects(
+        db.prepare(`SELECT * FROM ${tableName};`).all<any>("missing"),
+        (err: any) => {
+          strictEqual(err.cause.message, "column not found: missing");
+          return true;
+        }
+      );
+    });
+
+    test("when extracting a column from multiple rows", async function () {
+      const stmt = db.prepare(`SELECT * FROM ${tableName};`);
+      const { results, meta, error } = await stmt.all<any>("counter");
+      strictEqual(meta.txn, undefined);
+      strictEqual(error, undefined);
+      assert(meta.duration != null);
+      deepStrictEqual(results, [1, 2, 3, 4]);
+    });
+
+    test("when select all statement has a bound parameter", async function () {
+      await getDelay(500); // Just to keep request frequency down...
+      const stmt = db.prepare(`SELECT * FROM ${tableName} WHERE counter >= ?;`);
+      const { results } = await stmt.bind(3).all();
+      deepStrictEqual(results, [
+        { id: 3, counter: 3, info: "three" },
+        { id: 4, counter: 4, info: "four" },
+      ]);
+    });
+
+    test("when select all statement has only named bound parameters", async function () {
+      await getDelay(500); // Just to keep request frequency down...
+      const stmt = db.prepare(
+        `SELECT * FROM ${tableName} WHERE counter >= :counter;`
+      );
+      const { results } = await stmt.bind({ counter: 3 }).all();
+      deepStrictEqual(results, [
+        { id: 3, counter: 3, info: "three" },
+        { id: 4, counter: 4, info: "four" },
+      ]);
+    });
+
+    test("when query attempts to join across chain types", async function () {
+      const stmt = db.prepare(
+        `SELECT * FROM ${tableName}, healthbot_1_1 WHERE ${tableName}.counter == heathbot_1_1.counter;`
+      );
+      await rejects(stmt.all(), (err: any) => {
+        strictEqual(
+          err.cause.message,
+          "network mismatch: mix of testnet and mainnet chains"
+        );
+        return true;
+      });
+    });
+
+    test("when select statement with where returns empty", async function () {
+      const stmt = db.prepare(`SELECT * FROM ${tableName} WHERE false;`);
+      const { results, meta, error } = await stmt.all();
+      strictEqual(meta.txn, undefined);
+      strictEqual(error, undefined);
+      assert(meta.duration != null);
+      deepStrictEqual(results, []);
+    });
+
+    test("when a runtime error throws during a secondary wait", async function () {
+      const { meta } = await db
+        .prepare(
+          `INSERT INTO ${tableName} (id, counter, info) VALUES (1, ?1, ?2)`
+        )
+        .bind(5, "Bobby")
+        .all();
+      // TODO: Are we sure we don't want to use the cause pattern for these ones?
```

Comment:
This is an open question. I honestly _don't_ like it per se, unless the message includes the cause message as well.

Reply:
I actually dislike it enough that I'm glad we didn't do it here.